### PR TITLE
feat: Setup CircleCI for lint and test

### PR DESCRIPTION
**Pull request by chrisallo**
We have decided to use Circle CI as the main CI tool 
- Reduce cost from github
- Parity with other JS projects in the company

##### Changes

* Add circle CI
* Setup Lint, Test
* Update Jest minor version

#####  Todo/Seperate PR: 
* Remove GH-Action
* Setup coverage

[CORE-5381](https://sendbird.atlassian.net/browse/CORE-5381)

[CORE-5381]: https://sendbird.atlassian.net/browse/CORE-5381?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by AhyoungRyu** [comment]
@chrisallo It seems Hoon's commit is included in the commit list which is pushed like 1 year ago 😄

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+commands:
+  command-build_and_test:
+    steps:
+      - run:
+          name: UIKit for React build and test
+          command: |
+            npm install
+            npm run build
+            npm test
```
Comment (sravan-s) [inline .circleci/config.yml:13]: This is `yarn` tho :(

Comment (sravan-s) [reply .circleci/config.yml:13]: https://github.com/sendbird/sendbird-uikit-react/blob/main/.github/workflows/test-n-lint.yml#L24

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          no_output_timeout: 15m
+
+# Define a job to be invoked later in a workflow.
+# See: https://circleci.com/docs/2.0/configuration-reference/#jobs
+jobs:
+  job-lint_and_test:
+    docker:
+      - image: cimg/node:16.19.1
+    parallelism: 4
+    resource_class: large
```
Comment (sravan-s) [inline .circleci/config.yml:28]: cc @chrisallo  this okay?

**Comment by sravan-s** [comment]
Setup caching:
Screen Shot 2023-05-19 at 2 16 52 AM


Using cahe
Screen Shot 2023-05-19 at 2 17 28 AM

**Review by AhyoungRyu — APPROVED** [review]

**Pull request merged**